Packs/Rapid7InsightVMCloud/Integrations/Rapid7InsightVMCloud/Rapid7InsightVMCloud.py

```diff
@@ -0,0 +1,455 @@
+
+
+def get_asset_command(client: Client, asset_id: str) -> dict:
+    """
+    Returns the assessment and details of an asset (specified by id).
+
+    Args:
+        client (Client Object): Client object with request.
+        asset_id(str) : The identifier of the asset to retrieve the details for.
+
+    """
+
+    method = "GET"
+    endpoint = f"/v4/integration/assets/{asset_id}"
+    response = client.make_request(
+        method=method,
+        url_suffix=endpoint
+    )
+    if response:
+        return response
+    else:
+        return_error('no response')
+
+
+def search_assets_command(client: Client, hostname: None, page: str, size: str) -> CommandResults:
+    """
+    Returns the inventory, assessment, and summary details for a page of assets.
+
+    Args:
+        client (Client Object): Client object with request.
+        hostname: Search criteria for filtering assets returned.
+        page: The index of the page (zero-based) to retrieve
+        size: The number of records per page to retrieve.
+
+    """
+    args = demisto.args()
+    host_name = args.get('hostname')
+    params = {
+        "page": args.get("page"),
+        "size": args.get("size")
+    }
+    method = "POST"
+    endpoint = "/v4/integration/assets"
+
+    assets = f"asset.name CONTAINS '{host_name}'"
+    data = {
+        "asset": f"{assets}"
+
+    }
+    response = client.make_request(
+        method=method,
+        url_suffix=endpoint,
+        json_data=data,
+        params=params
+    )
+    if response:
+        response_data = response.get("data")
+        markdown = tableToMarkdown(
+            'This is required Asset data',
+            response_data
+        )
+        result = CommandResults(
+            readable_output=markdown,
+            outputs_prefix='Rapid7.InsighVMCloud.Assets',
+            outputs_key_field='id',
+            outputs=response_data
+        )
+        return result
+    else:
+        return_error('no response')
+
+
+def get_scan_command(client: Client, scan_id=str) -> CommandResults:
+    """
+    Retrieves the scan with the specified identifier.
+
+    Args:
+        client (Client Object): Client object with request.
+        scan_id : The identifier of the scan.
+
+    """
+
+    endpoint = f"/v4/integration/scan/{scan_id}"
+    method = 'GET'
+    response = client.make_request(
+        method=method,
+        url_suffix=endpoint
+    )
+    if response:
+        response_data = response
+        markdown = tableToMarkdown(
+            'This is the Required Scan Information',
+            response_data
+        )
+        result = CommandResults(
+            readable_output=markdown,
+            outputs_prefix='Rapid7.InsighVMCloud.Scans',
+            outputs_key_field='id',
+            outputs=response_data
+        )
+        return result
+    else:
+        return_error('no response')
+
+
+def get_scan_engines_command(client: Client, page: int, size: int) -> CommandResults:
```

Comment:
Why do we need the variables `age: int, size: int` they are not used ? @thimanshu474

Reply:
It is used in params variable in command.

Reply:
You use strings, not variables, so in that case the variables are unused

Reply:
To resolve this issue please edit these line.
    args = demisto.args()
    page = args.get("page")
    size = args.get("size")
    method = "GET"
    params = {
        "page": page,
        "size": size
    }

Reply:
It won't solver it the variables in the function are still redundant page for example will be `args.get("page")` if it exist or `None` if it's not and the page in the function signature is meaningless `search_vulnerabilities_command(client: Client, query: str, page: int, size=int)` why do you need the variables in the signature if you're getting them from args ?
@thimanshu474

Reply:
Please delete the variable taken from args and replace those line with below code in which directly taken value from function signature variables.

def get_scan_engines_command(client: Client, page: str, size: str)-> CommandResults:

    method = "GET"
    params ={
        "page": page,
        "size": size
    }
    endpoint = f"/v4/integration/scan/engine"
    response = client.make_request(method = method, url_suffix=endpoint, params=params)
    if int(size) > 500:
        return_error("Exceed size limit")
   `